drodrpg/DRODLib/Character.cpp

```diff
@@ -2482,14 +2491,17 @@ void CCharacter::Process(
 			case CCharacterCommand::CC_WaitForPlayerToMove:
 			{
 				//Wait until player moves in direction X.
-				if (!DidPlayerMove(command, player, nLastCommand))
+				if (bRoomBeingDisplayedOnly || !DidPlayerMove(command, player, nLastCommand))
 					STOP_COMMAND;
 
 				bProcessNextCommand = true;
 			}
 			break;
 			case CCharacterCommand::CC_WaitForPlayerToTouchMe:
 			{
+				if (bRoomBeingDisplayedOnly)
+					STOP_COMMAND;
```

Comment:
This doesn't work quite right -- the rest of the block will still execute. We need to do this:
`if (player.wX == this->wX && player.wY == this->wY && !bRoomBeingDisplayedOnly) ...`